kie-server-parent/kie-server-services/kie-server-services-jbpm/src/test/java/org/kie/server/services/jbpm/security/ElytronUserGroupAdapterTest.java

```diff
@@ -95,4 +106,43 @@ public void testSecurityForLoggedUser() {
                 .hasSize(3)
                 .contains(ROLE_1, ROLE_2, ROLE_3);
     }
+
+    @Test
+    public void testSecurityOnRealmUnavailable() throws RealmUnavailableException {
+        setAuthorizationFailureExceptionClass();
+        when(adapter.isActive()).thenCallRealMethod();
+        when(adapter.getGroupsForUser(Mockito.anyString())).thenCallRealMethod();
+        when(adapter.getUserName()).thenReturn(WRONG_USER_ID);
+        when(adapter.runAsPrincipalExists(USER_ID)).thenThrow(new RealmUnavailableException());
+        lenient().when(adapter.toPrincipalRoles(Mockito.anyString())).thenReturn(Arrays.asList(ROLE_1, ROLE_2, ROLE_3));
+        lenient().when(adapter.toRunAsPrincipalRoles(Mockito.anyString(), Mockito.eq(false))).thenReturn(Arrays.asList(ROLE_1, ROLE_2, ROLE_3));
+
+        List<String> roles = adapter.getGroupsForUser(USER_ID);
+
+        Assertions.assertThat(roles)
+                .isNotNull()
+                .hasSize(0);
```

Comment:
```suggestion
                .isEmpty();
```
from Sonar cloud suggestions to have it simplified ;)